### ISSUE TITLE
Gen 4 translations / txt files missing

### ISSUE THREAD

**Issue by gman-php** [ZeChrales/PogoAssets#14]
Hey Chrales,

not sure where the best place is to ask for it, but: Would it be possible to add the latest translations / TXT files, especially for all Gen 4 Pokemon recently added?

Thanks and greetings,

Andy!

**Comment by JayTurnr** [comment]
And Gen 4 image icon

**Comment by fratt** [comment]
@ZeChrales We need you man :'(

**Comment by ZeChrales** [comment]
https://github.com/ZeChrales/PogoAssets/blob/master/static_assets/txt/merged.txt

As I said multiple times already on discord, they have changed their way to manage texts. You will have to parse the file now ...

**Comment by hrk** [comment]
Thank you very much for the merged.txt file. As I only follow github and not discord, any hope/way to have the merged.txt files for the other available languages?

**Comment by gman-php** [comment]
Thx @ZeChrales! I haven't seen this on Discord. Do you have any hints on how we can easily process that file on our own then?

**Comment by gman-php** [comment]
> Thank you very much for the merged.txt file. As I only follow github and not discord, any hope/way to have the merged.txt files for the other available languages?

I would guess you have to collect them from the APKs,

**Comment by ZeChrales** [comment]
In https://github.com/ZeChrales/PogoAssets/blob/master/static_assets/txt you now have all merged.txt for all languages ... names are not great, but now it's automated !

**Comment by hrk** [comment]
Thank you very much!

**Comment by gman-php** [comment]
Any hints on how to process that file, e.g. on the command line and/or getting a tabbed/comma/json/some text format version out of it similar to the old ones?

**Comment by Sil3ntStorm** [comment]
While this is all good, I'm curious as to how one is supposed to identify which of the files pertains to which language?  
There doesn't appear to be any indicator for the language in the file itself and the number in the file name doesn't correspond to any meaningful / standard either (that I can see / find anyway).  
Are we just supposed to hardcode the file names? Are these expected to change when you update these assets in the future?

**Comment by JayTurnr** [comment]
@Sil3ntStorm Taking a quick look it seems you can figure this out via the `terms_of_service_url` string :+1: 

Hopefully, @ZeChrales will rename these files in future releases to be more clear, but we can only hope. I'm just glad to have all text files in one file!

**Comment by JayTurnr** [comment]
@Sil3ntStorm See #18

**Comment by Sil3ntStorm** [comment]
@JayTurnr nice find. That does seem like a good solution.
I don't even want to know how you managed to find that one or how long it took. Thanks a lot.

**Comment by JayTurnr** [comment]
Honestly, a few seconds. I identified English, searched for `en` and found a common string :+1:

**Comment by Sil3ntStorm** [comment]
That's not an uncommon term in English words. In fact it's still 858 results in translated values alone :cold_sweat:

**Comment by gman-php** [comment]
Thx @ZeChrales for adding those texts to the repo, already super helpful!
Thx @JayTurnr for the hints on parsing! Still, I would really appreciate if @ZeChrales may know what type of file that is, e.g. which language it should be. I really want to process / compile it better than just process it line by line and do regex on it ;) I have zero clue of Unity, do don't know if this is some kind of resource file. It doesn't seem to be a Unity Script, as this is neither C#, nor JS or Boo...

Very much appreciated!

**Comment by BrunnerLivio** [comment]
@gman-php do you mind open sourcing this? I will need to parse this file too

**Comment by gman-php** [comment]
@BrunnerLivio, sure :) But first we need to know what format that is. Then I can write sth to natively interpret/compile it and get a json/csv out of it

**Comment by BrunnerLivio** [comment]
@gman-php 
Do not know what format this is :/

For those of you who need a solution asap,
here is a regex for parsing key/value of a the merged-files:
```regex
/string Key = \"(.*?)\"\n.*?string Translation = \"(.*?)\"/gm
```

**Comment by Sil3ntStorm** [comment]
@gman-php I'd think this is the result of [Unity script serialization](https://docs.unity3d.com/Manual/script-Serialization.html)

**Comment by ryankane** [comment]
@BrunnerLivio Your regular expression will not work on this e.g.

```
			[1617]
			Entry data
				string Key = "player_cheating_warning|b"
				string Translation = "Warning Notice

We have detected activity on your account that indicates you are using modified client software or unauthorized third-party software in violation of our  Terms of Service.

This is the first Strike against your account. It will last for approximately {0} more days.

We strongly encourage you to stop this activity. Any further transgressions will result in an account suspension.

<a href="three_strike_policy_url">Visit the Help Center </a> for more information."
```

**Comment by ryankane** [comment]
@BrunnerLivio I updated my comment a few times.

@JayTurnr  I agree with the renaming of the "merged" text files.

**Comment by gman-php** [comment]
> @gman-php I'd think this is the result of [Unity script serialization](https://docs.unity3d.com/Manual/script-Serialization.html)

Good find actually. I'm trying to digg into that a little bit, especially in how to read these files outside unity. Maybe somebody has done something already... Will post if I find something!

**Comment by derigel23** [comment]
https://github.com/HearthSim/UnityPack

**Comment by ryankane** [comment]
@gman-php, it appears that the assets were renamed…

* `static_assets/txt/merged.txt`     – Portuguese (Brazilian)
* `static_assets/txt/merged #5.txt`  – Chinese
* `static_assets/txt/merged #6.txt`  – English
* `static_assets/txt/merged #10.txt` – French
* `static_assets/txt/merged #13.txt` – German
* `static_assets/txt/merged #16.txt` – Italian
* `static_assets/txt/merged #19.txt` – Japanese
* `static_assets/txt/merged #21.txt` – Korean
* `static_assets/txt/merged #24.txt` – Spanish

**Comment by gman-php** [comment]
I still couldn't figure out to properly read them with a tool, but I just use the RegEx from @BrunnerLivio and @ryankane. Guess there's no other way.
